specification.md

```diff
@@ -1,15 +1,279 @@
+verification.
+
 ### Generation
 
+#### SBOMit document
+To generate a SBOMit document, one must generate the consituent parts.  For the
+in-toto layouts, sub-layouts, attestations, and links, this process is 
+described in the in-toto project's documentation.  Note however, that for the 
+purposes of SBOMit, the layout is required to include certain information 
+so that this information may be used to populate the SIT.  
+
+Generating the mutator by hand may done by starting with a null mutator and 
+then using tooling to generate a SIT in the correct format.  The SIT file may 
+be modified and the JSON diff may be computed.
+
+Similarly, if a SBOMit document is modified, one may elect to add an addendum
+to modify it instead of generating a fresh document.  This is easily done by
+generating the new SBOMit document and then creating a JSON diff.
+
+#### SIT 
+
+The SIT is generated by first deriving common information from in-toto and
+placing it in an intermediate format called X.  This format is effectively
+a SBOM itself which matches the NTIA format....
+
+Then the tool specified in the layout may be used to derive the SIT in the
+appropriate end format.  Note also, that multiple SITs may be generated from 
+the same SBOMit document.  So, for example, one SBOMit document could generate
+both a SPDX SIT and a CycloneDX SIT.
+
+An important part of generating the SIT is applying the mutator to the output
+of the tool used to derive the SIT.  This adds or modifies information in the 
+resulting SIT as instructed.  This works as though applying a JSON patch to 
+the SIT document.
+
+It is recommended that the SIT generation tool provides a warning if the
+mutator modifies (e.g., replaces or removes) portions of the SIT derived from
+the in-toto information.  This is becuase this will be flagged as a security
+risk when verification of the SBOMit document is performed.
+
 ### Verification
+
+#### SBOMit document
+
+SBOMit document verification proceeds by the following steps:
+
+* verifying the in-toto layout over the provided in-toto metadata.  
+* checking each SIT to ensure they are signed by the correct SIT generator
+* generate each SIT using the relevant tooling to examine the mutator 
+  behavior.  Any parts of the SIT that the mutator has chosen to overwrite or 
+  remove represent a substantial security risk.  These are specifically 
+  flagged during validation.  The user must specify a flag like 
+  --insecure-allow-in-toto-override or respond to a similarly scary alert to 
+  install the software.
+
+If this matches the generated SITs, then the SBOMit document verification
+has succeeded.
+
+#### SIT 
+
+A SIT may be verified in two ways.  
+
+First, the SBOMit document may be downloaded and then verified.  This provides 
+strong security guarantees, but is a fairly heavyweight operation.
+
+Second, the SIT may have its signature checked, much like any signed SBOM.
```

Comment:
How is the key for this distributed?

Reply:
I'm imagining the same as you would usually for an SBOM.  I'm actually not completely clear how this process works in every case today.  I imagine it is like in-toto layout keys, where they are just assumed to exist on the client, but I would love to learn the "correct" way, if I'm wrong.